### ISSUE TITLE
Icons in PL app CLI output broken in some terminals

### ISSUE THREAD

**Issue by awaelchli** [Lightning-AI/lightning#14226]
### To Reproduce

```
lightning init pl-app train.py
```

![image](https://user-images.githubusercontent.com/5495193/184859636-85e6c35b-6f84-4b36-a077-c21e5b22e4d4.png)

### Expected behavior

Icons get displayed or hidden

The fix would be similar as in [https://github.com/PyTorchLightning/pytorch-lightning/blob/dd475183227644a8d22dca3[…]fb0a9b2c4/pytorch_lightning/loops/dataloader/evaluation_loop.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/dd475183227644a8d22dca3[%E2%80%A6]fb0a9b2c4/pytorch_lightning/loops/dataloader/evaluation_loop.py)

### Environment

This is a copy of an old [linear ticket](https://linear.app/lightning-ai/issue/LAI2-9250/the-icons-in-the-summary-dont-get-displayed-correctly) from user testing.

**Comment by bpkapkar** [comment]
This link is not working.

https://github.com/Lightning-AI/lightning/blob/dd475183227644a8d22dca3%5B%E2%80%A6%5Dfb0a9b2c4/pytorch_lightning/loops/dataloader/evaluation_loop.py

![image](https://user-images.githubusercontent.com/49440262/184872373-5ca46af0-b6ea-4954-bebb-b12f894859da.png)

**Comment by awaelchli** [comment]
It is here: https://github.com/Lightning-AI/lightning/blob/2622989b108368813fdd3850b002fd1ad69988c2/src/pytorch_lightning/loops/dataloader/evaluation_loop.py#L379-L385

**Comment by Rajathbharadwaj** [comment]
Can I contribute

**Comment by awaelchli** [comment]
@Rajathbharadwaj Feel free to give it a try :)

**Comment by awaelchli** [comment]
The relevant code where this needs to be fixed is here:

https://github.com/Lightning-AI/lightning/blob/8662ab25d4e32196bbb4489100ca46c3fa885248/src/lightning_app/cli/cmd_pl_init.py#L151-L153

**Comment by Rajathbharadwaj** [comment]
If I understand correctly, I need to add 

```
     if sys.stdout.encoding is not None: 
         "─".encode(sys.stdout.encoding) 
 except UnicodeEncodeError: 
     bar_character = "-" 
 else: 
     bar_character = "─" 
```

to this right

```
 icon = "📂 " if path.is_dir() else "📄 " 
 tree.add(Text(icon) + text_pathname) 
  
```

**Comment by awaelchli** [comment]
Almost 😅 
It needs to be something like this:

```py
try:
    # try encoding the icon
    icon = ... # proper icon
 except UnicodeEncodeError: 
     icon = ""  # no icon
 else: 
     ...
```

**Comment by christopher-nguyen-re** [comment]
I saw that this issue is still open. Would it be alright if I take a crack at it? I am new to contributing to open source software but am willing to try!

**Comment by awaelchli** [comment]
Yes @christopher-nguyen-re. I don't think @Rajathbharadwaj has started yet. Feel free to give it a try. When you open a PR, we can also help finish it or help fix anything that is giving trouble. Cheers and thanks for your interest 😃

**Comment by christopher-nguyen-re** [comment]
I attempted to replicate the bug running `lightning init pl-app train.py`. My terminal does properly show the icons. Could you elaborate on what terminal you used? I will continue working on the bug fix in the meantime.
